core/src/main/java/org/eclipse/hono/util/RegistrationConstants.java

```diff
@@ -158,17 +160,20 @@ public static JsonObject getRegistrationJson(final String action, final String t
     }
 
     public static JsonObject getRegistrationJson(final String action, final String tenantId, final String deviceId, final JsonObject payload) {
-        return getRegistrationJson(action, tenantId, deviceId, null, payload);
+        return getRegistrationJson(action, tenantId, deviceId, null, null, payload);
     }
 
-    public static JsonObject getRegistrationJson(final String action, final String tenantId, final String deviceId, final String key, final JsonObject payload) {
+    public static JsonObject getRegistrationJson(final String action, final String tenantId, final String deviceId, final String key, final String value, final JsonObject payload) {
         final JsonObject msg = new JsonObject();
         msg.put(APP_PROPERTY_ACTION, action);
         msg.put(APP_PROPERTY_DEVICE_ID, deviceId);
         msg.put(APP_PROPERTY_TENANT_ID, tenantId);
         if (key != null) {
             msg.put(APP_PROPERTY_KEY, key);
         }
+        if (value != null) {
```

Comment:
No, because that is the current implementation. Complex values are allowed and the code does run into the ClassCastException. I propose to ignore complex values in search, so we do not run into the exception. The other solution would be to not allow complex values at all.